firefox_ui_harness/runners/update.py

```diff
@@ -0,0 +1,76 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import shutil
+import tempfile
+
+import mozfile
+import mozinstall
+
+from base import FirefoxUITestRunner
+from ..testcases import UpdateTestCase
+
+
+DEFAULT_PREFS = {
+    'app.update.log': True,
+}
+
+
+class UpdateTestRunner(FirefoxUITestRunner):
+
+    def __init__(self, **kwargs):
+        FirefoxUITestRunner.__init__(self, **kwargs)
+
+        self.original_bin = self.bin
+
+        self.prefs.update(DEFAULT_PREFS)
+
+        # In case of overriding the update URL, set the appropriate preference
+        override_url = kwargs.pop('update_override_url', None)
+        if override_url:
+            self.prefs.update({'app.update.url.override': override_url})
```

Comment:
Some of the recent changes for preference handling broke the override url handling of Firefox. Not sure why, but even with this preference set, the `app.update.url` pref is not getting updated.